src/deluge/model/drum/drum.h

```diff
@@ -68,7 +68,7 @@ class Drum {
 	virtual void noteOff(ModelStackWithThreeMainThings* modelStack, int32_t velocity = kDefaultLiftValue) = 0;
 	virtual bool allowNoteTails(ModelStackWithSoundFlags* modelStack, bool disregardSampleLoop = false) = 0;
 	virtual bool anyNoteIsOn() = 0;
-	virtual bool hasAnyVoices() = 0;
+	virtual bool hasAnyVoices(bool resetTimeEntered) = 0;
```

Comment:
The reset logic should probably go in a separate function

Reply:
Yeah reasonable. I copied the synth approach but the drum check for voices is a lot simpler

Reply:
done